solutions/bio2/TerminalVelocity/index.tsx

```diff
@@ -0,0 +1,47 @@
+import { Link, SCode, SCodeBlock, SList, SText, STitle } from "components";
+import { brace, op } from "../../util";
+
+export const TerminalVelocity = () => {
+  return (
+    <>
+      <STitle>Terminal Velocity</STitle>
+
+
+      <STitle>Abridged Statement</STitle>
+      <SText>
+        This problem is obscured by a lot of legend. Note that two pigeons have a crossing flight path 
+        if and only if the one with a larger starting position (competitor number) comes before the one 
+        with the smaller starting position. This leads us to this formalised statement: Given all the 
+        inversions of a permutation, reconstruct a possible permutation. 
+        <a href="https://en.wikipedia.org/wiki/Inversion_(discrete_mathematics)">(definition of inversions)</a>
+      </SText>
+
+      <STitle>Idea</STitle>
+      <SText>
+        Compute for each pigeon how many pigeons come before it in the final permutation.
+      </SText>
+
+      <STitle>Solution</STitle>
+      <SText>
+        Since we are given all of the inversions, for a particular pigeon $x$ we know exactly which other 
+        pigeons will come before it in the final permutation. These are exactly the pigeons $y$ with a smaller 
+        starting position $y<x$ such that $(x,y)$ is not an inversion, and the pigeons $y$ with a larger starting 
+        position $y>x$ such that $(x,y)$ is an inversion.
+      </SText>
+      <SText>
+        Therefore, for each $x$ we can count the number of pigeons which come before $x$ in the final ordering. 
+        Let's process the inversion pairs one by one. At the beginning, with no inversions processed, the final 
+        position is the same as the initial position. Then, when processing pair $(x,y)$, we swap $x$ and $y$ if $y\x$ 
+        to ensure that $x<y$. Now we know that one more pigeon comes before $x$ than we thought before (pigeon $y$), 
+        and we know that one less pigeon comes before $y$ than we thought before (pigeon $x$), so we can update our counts. 
+        In this way we know for each pigeon $i$ how many pigeons $j$ come before $i$ in the final ordering, and thus we 
+        know the position of each initial pigeon in the final ordering.
+      </SText>
+
+      <STitle>Time Complexity</STitle>
```

Comment:
I don't think this needs to be under a separate heading. Just put it at the end of the previous section.